### PR TITLE
Site block: improve title and change misleading aria-label

### PR DESCRIPTION
**Pull request by bluefuton**
The `Site` block helpfully provides an `aria-label` for screenreader users, but its only possible values are:
- Open site %(domain)s in new tab
- Open site %(domain)s in a preview

Now the block is used more widely, those two labels do not cover all of the possible actions. For example, in Reader Share, we use `SiteSelector` and it ends up rendering this:

screen shot 2017-08-10 at 13 25 46

In this scenario, we are not opening bluefutontest5.wordpress.com in a new tab - we're selecting it to reblog a post to.

To support the wider range of uses for `Site`, this PR improves the `title` attribute. It now includes the site name:

Before: "View this site"
After: "View site bluefutontest5.wordpress.com"

Before: "Select this site"
After: "Select site bluefutontest5.wordpress.com"

... and uses the same values for `aria-label`, since `title` is sometimes unreliable on assistive technologies.

Would it also help to have a custom aria-label specified as a prop? Feedback most welcome.

### To test

Visit http://calypso.localhost:3000/devdocs/blocks/site and check the `title`/`aria-label` on the Site block examples.

### CODE REVIEW

**Comment by bluefuton** [comment]
Tests are failing because of a lint error. It's a disagreement between eslint and prettier and can be safely disregarded 👍

File: client/blocks/site/index.jsx [hunk 1]
```diff
@@ -1,18 +1,21 @@
+/** @format */
 /**
  * External dependencies
  */
 import React from 'react';
-import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
```
Comment (samouri) [inline client/blocks/site/index.jsx]: nice

File: client/blocks/site/index.jsx [hunk 2]
```diff
@@ -88,24 +91,31 @@ const Site = React.createClass( {
...
-						} )
-						: this.translate( 'Open site %(domain)s in new tab', {
-							args: { domain: site.domain }
-						} )
+					aria-label={
+						this.props.homeLink
+							? translate( 'View site %(domain)s', {
+									args: { domain: site.domain },
+								} )
+							: translate( 'Select site %(domain)s', {
```
Comment (samouri) [inline client/blocks/site/index.jsx]: thanks for fixing this lie :)

**Comment by samouri** [comment]
Code LGTM. Testing now

**Comment by blowery** [comment]
@samouri how'd this test out?

**Comment by samouri** [comment]
@blowery: i kept _almost_ testing it out and then getting distracted by conversations about conversations tool.  

just tested right now: tests well :)

**Pull request merged**